errai-codegen/src/main/java/org/jboss/errai/codegen/meta/impl/apt/APTMethod.java

```diff
@@ -67,7 +67,7 @@ public MetaClass getReturnType() {
       return new APTClass(((Type.MethodType) typeMirror).getReturnType());
     }
 
-    return new APTClass(method.getReturnType());
+    return new APTClass(method.getReturnType()).getErased();
```

Comment:
Do we know if all the MetaMethod implementations return an erased type?

Reply:
Same as below.

Reply:
Okay, I'm assuming we don't have test coverage for this behaviour. Would you mind adding tests for this (and similar behaviour in MetaParameter) to the AbstractMetaClassTest?

Reply:
There were tests for the return type of a MetaMethod, but not for MetaParameters. I added them with the same comment referencing [ERRAI-459](https://issues.jboss.org/browse/ERRAI-459). It seems like this behavior is there for a long time now.. This is certainly something that could be addressed in the future, but I don't see it as being a part of this migration we're in the middle of, so if it's okay for you, I'll not deal with that in this PR.

Reply:
That's fine. I'm good to merge this PR when the build finishes.